multiversx_sdk_cli/dependencies/modules.py

```diff
@@ -293,73 +314,27 @@ def uninstall(self, tag: str):
         if os.path.isdir(directory):
             shutil.rmtree(directory)
 
-    def is_installed(self, tag: str) -> bool:
-        resolution = self.get_resolution()
+    def _is_installed_and_set_to_nightly(self) -> bool:
+        # the method parameter is not used in this specific module
+        is_rust_installed = self.is_installed("")
 
-        if resolution == DependencyResolution.Host:
-            which_rustc = shutil.which("rustc")
-            which_cargo = shutil.which("cargo")
-            logger.info(f"which rustc: {which_rustc}")
-            logger.info(f"which cargo: {which_cargo}")
-
-            return which_rustc is not None and which_cargo is not None
-        if resolution == DependencyResolution.SDK:
-            try:
-                env = self._get_env_for_is_installed_in_sdk()
-                myprocess.run_process(["rustc", "--version"], env=env)
-                return True
-            except Exception:
-                return False
+        if not is_rust_installed:
+            return is_rust_installed
+        else:
+            self._recommend_default_rust_version()
```

Comment:
Additionally, you can directly inline the implementation of `_recommend_default_rust_version` inside `install` - also because the warning message mentions the "overwrite" flag, and the overwrite check is performed there. 

Can also stay as it is.

Reply:
Moved it inside the `install()` method.